Notesnook.API/Controllers/MonographsController.cs

```diff
@@ -206,16 +218,61 @@ public async Task<IActionResult> TrackView([FromRoute] string id)
             if (monograph == null) return Content(SVG_PIXEL, "image/svg+xml");
 
             if (monograph.SelfDestruct)
-                await Monographs.DeleteByIdAsync(monograph.Id);
+            {
+                var userId = this.User.FindFirstValue("sub");
+                await Monographs.Collection.UpdateOneAsync(
+                    CreateMonographFilter(userId, new Monograph { ItemId = id }),
+                    Builders<Monograph>.Update
+                    .Set(m => m.Deleted, true)
+                );
+                SyncMonograph(id, null);
+            }
 
             return Content(SVG_PIXEL, "image/svg+xml");
         }
 
         [HttpDelete("{id}")]
-        public async Task<IActionResult> DeleteAsync([FromRoute] string id)
+        public async Task<IActionResult> DeleteAsync([FromQuery] string deviceId, [FromRoute] string id)
         {
-            await Monographs.Collection.DeleteOneAsync(CreateMonographFilter(id));
+            var userId = this.User.FindFirstValue("sub");
+            await Monographs.Collection.UpdateOneAsync(
+                    CreateMonographFilter(userId, new Monograph { ItemId = id }),
+                    Builders<Monograph>.Update
+                    .Set(m => m.Deleted, true)
+            );
+            SyncMonograph(id, deviceId);
             return Ok();
         }
+
+        private void SyncMonograph(string monographId, string deviceId)
+        {
+            var userId = this.User.FindFirstValue("sub");
+            var jti = this.User.FindFirstValue("jti");
+
+            Task.Run(async () =>
+            {
+                if (deviceId == null)
+                {
+                    var emptyString = string.Empty;
+                    await new SyncDeviceService(new SyncDevice(ref userId, ref emptyString)).AddIdsToAllDevicesAsync(new List<string> { $"{monographId}:monograph" });
+                }
```

Comment:
I don't think we need this: people on older versions shouldn't expect for monographs to sync.

Reply:
This is for the case where monograph self destructs where we have to mark the monograph id as unsynced for all devices. I should clarify this intention in the code

Reply:
It will also get called when someone from an older client tries to publish or update a monograph.